inventory_management_system_api/services/catalogue_category_property.py

```diff
@@ -89,7 +94,21 @@ def create(
             stored_catalogue_category.catalogue_item_properties + [catalogue_item_property]
         )
 
-        catalogue_item_property_in = CatalogueItemPropertyIn(**catalogue_item_property.model_dump())
+        # Check unit id exists in units collection
+        unit = None
+        # pylint: disable=duplicate-code
+        if catalogue_item_property.unit_id is not None:
+            unit_id = catalogue_item_property.unit_id
+            unit = self._unit_repository.get(unit_id)
+            if not unit:
+                raise MissingRecordError(f"No unit found with ID: {unit_id}")
+        # pylint: enable=duplicate-code
+
+        catalogue_item_property_in = CatalogueItemPropertyIn(
+            **{**catalogue_item_property.model_dump(), "unit": unit.value}
+        )
```

Comment:
This line causes a 500 error for the request
```
{
	"name": "test5",
	"type": "number",
	"mandatory": true,
	"allowed_values": null,
	"default_value": 0
}
```
as unit is None. We should have an e2e test to ensure this gets picked up.